### ISSUE TITLE
Saplings

### ISSUE THREAD

**Issue by IntegratedQuantum** [PixelGuys/Cubyz#546]
Saplings should be able to turn into structures.
Saplings should be obtainable from trees at a rate that does not allow you to farm it infinitely.

**Comment by IntegratedQuantum** [comment]
Since there is an open PR for this now in #2585, I would like to have some input on whether or not we want terraria style saplings (in the inventory they are represented by their seed, e.g. an acorn, and only when you plant them they turn into a little bush) or minecraft style saplings (the sapling is an item)

**Comment by DellieDelta** [comment]
I like the seed option most! Also makes sense that leaves drop acorns instead of saplings

**Comment by careeoki** [comment]
I would do seeds.

**Comment by Argmaster** [comment]
I think seeds are better.

**Comment by careeoki** [comment]
How I would do acorns/pinecones/etc is to have them as blocks hanging from trees, which then can be harvested with a sickle. 
Not every oak tree would have acorns, but ones that do have 3-4. So when growing trees there's a chance you won't get any acorns at all.

Question though, some of our trees would have fruits rather than acorns or seeds or whatever (coconuts for palm trees, baobab fruit for baobab trees)
If we want to use them in food prep, it would make sense to have them grow on trees like a crop so you have lots to cook with. But if the fruit is also the sapling item, then you have infinite wood.

**Comment by Wbjitscool** [comment]
yeah seeds are a better way to do it like this

**Comment by IntegratedQuantum** [comment]
> If we want to use them in food prep, it would make sense to have them grow on trees like a crop so you have lots to cook with. But if the fruit is also the sapling item, then you have infinite wood.

That is a good point.
We could make saplings a separate step from seeds, maybe you need a pot, some soil, and 10 seeds to make one sapling, or maybe it requires some external resource, like a "growth crystal" which would limit it from another angle.

**Comment by Argmaster** [comment]
Is the vendetta against renewable resources worth adding more complicated process than even Vintage Story has?
Just make the seed grow into a tree with X% chance to just disappear instead.

**Comment by Wbjitscool** [comment]
?

**Comment by Wbjitscool** [comment]
> Is the vendetta against renewable resources worth adding more complicated process than even Vintage Story has? Just make the seed grow into a tree with X% chance to just disappear instead.

wait kinda like how you loose some seeds whn trying to grow some and they die but some survive is that what you mean?

**Comment by IntegratedQuantum** [comment]
> Just make the seed grow into a tree with X% chance to just disappear instead.

I though about this, but I think it would be annoying if you want to have a tree in a specific place.

**Comment by ZigguratV** [comment]
Would having the crystal force seed to always sprout into a tree while having the option to gamble if the seed sprouts or not work?
Also how would fruit trees work?

**Comment by Argmaster** [comment]
> I though about this, but I think it would be annoying if you want to have a tree in a specific place.

Then add a dead and not dead plant thingy that pops up immediately when planting with X% to be the not-dead variant. One dead plant should be sufficient, sprouting plants are more or less generic.

**Comment by careeoki** [comment]
Maybe saplings will only grow into a tree if they are placed on a rich soil/compost block, and when they grow, they consume the nutrients and convert it back into regular soil. Then you have to craft more compost before growing more trees.

**Comment by ikabod-kee** [comment]
Hey, that sounds pretty good. Block interaction :)

**Comment by careeoki** [comment]
This way you could have a thousand coconuts but are still limited by the amount of compost you have.

**Comment by ikabod-kee** [comment]
I can see compost being an "ore" that you place on soil.

**Comment by IntegratedQuantum** [comment]
How would you craft compost then?

**Comment by tillpp** [comment]
2 stacks of leaf 🗿
(this is a joke)

**Comment by IntegratedQuantum** [comment]
It would have to be more than that even

**Comment by ikabod-kee** [comment]
I have an idea: Why not use sulfur? I mean, it works as a good compost in real life!

**Comment by careeoki** [comment]
> How would you craft compost then? / It would have to be more than that even

Ideally there should be a composter where you can input any unwanted plant matter and eventually get some compost out of it (maybe with the help of some worms)
But for now it could be like 120 grass vegetation + 10 mushrooms + 10 sulfur (grass as fodder material, mushrooms to help break down materials, sulfur as fertilizer and to make it more expensive)

> I have an idea: Why not use sulfur? I mean, it works as a good compost in real life!

Our sulfur ore is large crystalline chunks, I think it should have to be processed in some way first.

**Comment by ikabod-kee** [comment]
> Our sulfur ore is large crystalline chunks, I think it should have to be processed in some way first.

Is crafting not an act of processing?

**Comment by careeoki** [comment]
Do you mean instead of a compost ore block, we just use sulfur ore to grow trees, or do you just mean using sulfur in the crafting recipe for compost?

**Comment by ikabod-kee** [comment]
Using sulfur in the crafting recipe!!

**Comment by careeoki** [comment]
Oh, okay, we agree then.

**Comment by Argmaster** [comment]
> 120 grass vegetation + 10 mushrooms + 10 sulfur

IMO large numbers are just grindy and boring. It would be better to require lets say 3-5 items of 3-5 **different** plants + sulfur to make compost block. We want people to explore, not to bore to death left clicking grass. With a little of thinking we can ensure players will have to visit multiple biomes to do that. S

toping players from doing something by just requiring large quantities is bard game design, it encourage boring grind instead of exploration. I would also be careful with the recipes for compost, so they don't involve multiple rare resources at once (eg. no cold vegetation and cactus flower in one thing) so they are not too frustrating to make. Ideally there should be some overlap between them.

Mind that intermediate processing products that exist only to make the process have multiple steps w/o alternative use are not too fun either. I would rather require a dedicated workstation over having complicated process.

**Comment by careeoki** [comment]
Yes, I want to have a composter where you can use any variety of plants to make your compost. I'm trying to figure out something that wouldn't require that since we are nowhere near being able to make a composter.

**Comment by Argmaster** [comment]
Well we are not that far, but yeah, it has to wait. We can have multi ingredient inventory recipes, only small issue would be to optimize how recipes with 5+ ingredients are displayed, but splitting ingredients into rows of 3 should do the trick

**Comment by IntegratedQuantum** [comment]
I also had a different idea in regards to seeds:
Most tree seeds for current trees aren't even edible, with the exception of coconuts and baobab fruits, both of which are pretty large, so I think it would make sense to just allow you to cut them into many servings, approaching the problem from the other side.

**Comment by careeoki** [comment]
I had thought of that too, it makes sense, it just puts apples in a weird place (weirder than it already is)

**Comment by IntegratedQuantum** [comment]
I think for ordinary fruits like apples we should probably just do the terraria thing, where each tree type can drop various different fruits.

**Comment by ZigguratV** [comment]
Could there be gardening focused on the same tree producing more fruits? Perhaps with some mechanic to splice existing tree with a sapling of a different one to have more dense fruit creation. Such fruits would have to be infertile if requirement for particularly enhanced soil isn't how the tree renewability will be limited
